pkg/appmanager/resourceConfig.go

```diff
@@ -919,6 +994,18 @@ func createRSConfigFromRoute(
 		rsCfg.HandleRouteTls(tls, pStruct.protocol, policyName, rule)
 	}
 
+	if pStruct.protocol == "http" && nil != tls &&
+		tls.Termination == routeapi.TLSTerminationEdge &&
+		tls.InsecureEdgeTerminationPolicy == routeapi.InsecureEdgeTerminationPolicyRedirect {
+		// TLS config indicates to forward http to https.
+		path := "/"
+		if route.Spec.Path != "" {
+			path = route.Spec.Path
+		}
+		svcFwdRulesMap.AddEntry(route.ObjectMeta.Namespace, route.Spec.To.Name,
+			route.Spec.Host, path)
+	}
+
```

Comment:
Would this make more sense to put into HandleRouteTls?

Reply:
HandleRouteTls isn't being called for updates (I'm not sure why) so if it was there we wouldn't handle changes to tls.InsecureEdgeTerminationPolicy properly.